File: Stack/Simplify Path/SimplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = deque()
        components = path.split("/")
        for component in components:
            if component == "..":
                if stack:
                    stack.pop()
            elif component and component != ".":
                stack.append(component)
        simplified_path = "/" + "/".join(stack)
        return simplified_path if simplified_path else "/"

File: Stack/Simplify Path/SimplifyPath.cs
public class Solution {
    public string SimplifyPath(string path) {
        Stack<string> stack = new Stack<string>();
        string[] components = path.Split('/');
        
        foreach (string component in components) {
            if (component == "..") {
                if (stack.Count > 0) {
                    stack.Pop();
                }
            } else if (!string.IsNullOrEmpty(component) && component != ".") {
                stack.Push(component);
            }
        }
        
        string[] simplifiedPathArray = stack.ToArray();
        Array.Reverse(simplifiedPathArray);
        string simplifiedPath = "/" + string.Join("/", simplifiedPathArray);
        
        return string.IsNullOrEmpty(simplifiedPath) ? "/" : simplifiedPath;
    }
}